EIPS/eip-4750.md

```diff
@@ -58,41 +58,37 @@ If the code is legacy bytecode, any of these instructions results in an *excepti
 
 First we define several helper values:
 
-- `caller_stack_height = return_stack.top().stack_height` - stack height value saved in the top item of return stack
 - `type[i].inputs = type_section_contents[i * 4]` - number of inputs of ith section
 - `type[i].outputs = type_section_contents[i * 4 + 1]` - number of outputs of ith section
+- `type[i].max_stack_height = type_section_contents[i * 4 + 2:i * 4 + 4]` - maximum operand stack height of ith section
 
 If the code is valid EOF1, the following execution rules apply:
 
 #### `CALLF`
 
 1. Has one immediate argument,`target_section_index`, encoded as a 16-bit unsigned big-endian value.
-2. EOF validation guarantees that operand stack has at least `caller_stack_height + type[target_section_index].inputs` items.
-3. If operand stack size exceeds `1024 - type[target_section_index].max_stack_height` (i.e. if the called function may exceed the global stack height limit), execution results in exceptional halt. This also guarantees that the stack height after the call is within the limits.
+3. If operand stack size exceeds `1024 - type[target_section_index].max_stack_height + type[target_section_index].inputs` (i.e. if the called function may exceed the global stack height limit), execution results in exceptional halt. This also guarantees that the stack height after the call is within the limits.
```

Comment:
I would leave point 2, with some rephrasing, it was added already with stack validation in mind (https://github.com/ethereum/EIPs/pull/6261/files)
```suggestion
2. EOF validation guarantees that operand stack has enough items to use as callee's inputs.
3. If operand stack size exceeds `1024 - type[target_section_index].max_stack_height + type[target_section_index].inputs` (i.e. if the called function may exceed the global stack height limit), execution results in exceptional halt. This also guarantees that the stack height after the call is within the limits.
```